### PR TITLE
Adds Deposit-Only Collection setting to CollectionType (#1385).

### PR DESCRIPTION
**Pull request by bwatson78**

### CODE REVIEW

**Comment by bwatson78** [comment]
Screen Shot 2020-08-25 at 10 59 02 AM

**Comment by bwatson78** [comment]
Screen Shot 2020-08-25 at 11 02 56 AM

**Comment by bwatson78** [comment]
@devanshu-m My first commit used the initializer/`class_eval` approach and produced errors because, somehow, the original `prepends` initializer wasn't operating correctly. The errors I was receiving stated that `low_res' was an unknown argument. My theory was, somehow, the amount of initializers was too much for the system and to test that, bring over the actual Hyrax classes I was trying to `class_eval` and just make the changes there. That theory worked.
For the other question, if you check the db migration, not only did I put `default: false`, but I also included `null: false`, which will automatically fill in `false` to all existing CollectionType instances.

File: app/views/hyrax/admin/collection_types/_form_settings.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<p><%= t('.instructions') %></p>
```
Comment (devanshu-m) [inline app/views/hyrax/admin/collection_types/_form_settings.html.erb:1]: Is this an overwrite? Can you please mention.

Comment (bwatson78) [reply app/views/hyrax/admin/collection_types/_form_settings.html.erb:3]: Last commit addresses the comment request.

**Pull request merged**